lib/src/main/scala/code/chymyst/jc/Reaction.scala

```diff
@@ -190,10 +202,10 @@ final case class OutputMoleculeInfo(molecule: Molecule, flag: OutputPatternType)
 }
 
 // This class is immutable.
-final case class ReactionInfo(inputs: List[InputMoleculeInfo], outputs: List[OutputMoleculeInfo], guardPresence: GuardPresenceFlag, sha1: String) {
+final case class ReactionInfo(inputs: Array[InputMoleculeInfo], outputs: Array[OutputMoleculeInfo], guardPresence: GuardPresenceFlag, sha1: String) {
 
   // The input pattern sequence is pre-sorted for further use.
-  private[jc] val inputsSorted: List[InputMoleculeInfo] = inputs.sortBy { case InputMoleculeInfo(mol, flag, sha) =>
+  private[jc] val inputsSorted: List[InputMoleculeInfo] = inputs.sortBy { case InputMoleculeInfo(mol, _, flag, sha) =>
```

Comment:
InputsSorted is pre-sorted so that it is traversed sequentially in the sorted order. Also, its traversal contains pattern-matching on List. That's why it's a list rather than an array. Everything else is an Array, because it's computed once at compile time and never going to change. (Although Array is mutable - I wish Array were immutable...)